README
NAME
    Dist::Zilla::PluginBundle::SHANTANU - Dist Zilla Plugin Bundle the way I
    like to use it

VERSION
    version 0.19

SYNOPSIS
       # in dist.ini
         [@SHANTANU]

DESCRIPTION
    This is a Dist::Zilla PluginBundle. The way I use it. While this bundle
    is customized to my needs, others might be better of forking this
    repository and modifying it to their own needs or using the more mature
    Plugin bundles that this is derived from like the one by David Golden.

ATTRIBUTES
  no_git
    makemaker attribute By default uses [MakeMaker::Awesome] This can be
    overriden by defining this attribute

  skip_makemaker
    Skip Default Makemaker option to add your own plugin for generating
    makefile

  no_git
    no_git attribute

  version_regexp
    version_regexp attribute

  exclude_filename
    list of filenames to exclude e.g. exclude_filename=dist.ini
    exclude_filename=META.json exclude_filename=META.yml

  exclude_match
    list of regex paths to exclude e.g. exclude_match=^inc\/.*

  stopwords
    Stopwords to exclude for spell checks in pod

USAGE
    To use this PluginBundle, just add it to your dist.ini. You can provide
    the following options:

    *   "is_task" -- this indicates whether TaskWeaver or PodWeaver should
        be used. Default is 0.

    *   "auto_prereq" -- this indicates whether AutoPrereq should be used or
        not. Default is 1.

    *   "tag_format" -- given to "Git::Tag". Default is 'release-%v' to be
        more robust than just the version number when parsing versions for
        "Git::NextVersion"

    *   "version_regexp" -- given to "Git::NextVersion". Default is
        '^release-(.+)$'

    *   "fake_release" -- swaps FakeRelease for UploadToCPAN. Mostly useful
        for testing a dist.ini without risking a real release.

    *   "weaver_config" -- specifies a Pod::Weaver bundle. Defaults to
        @SHANTANU.

    *   "stopwords" -- add stopword for Test::PodSpelling (can be repeated)

    *   "no_git" -- bypass all git-dependent plugins

    *   "no_critic" -- omit Test::Perl::Critic tests

    *   "no_spellcheck" -- omit Test::PodSpelling tests

    *   "no_coverage" -- omit Pod Coverage tests

    When running without git, C<GatherDir> is used instead of
    C<Git::GatherDir>, C<AutoVersion> is used instead of
    C<Git::NextVersion>, and all git check and commit operations are
    disabled.

    This PluginBundle now supports ConfigSlicer, so you can pass in options
    to the plugins used like this:

       [@SHANTANU]
       ExecDir.dir = scripts ; overrides ExecDir

COMMON PATTERNS
  use github instead of RT
       [@SHANTANU]
       :version = 0.32
       AutoMetaResources.bugtracker.github = user:shantanu
       AutoMetaResources.bugtracker.rt = 0

SEE ALSO
    *   Dist::Zilla

    *   Dist::Zilla::Plugin::PodWeaver

    *   Dist::Zilla::Plugin::TaskWeaver

SUPPORT
  Bugs / Feature Requests
    Please report any bugs or feature requests through github at
    <https://github.com/shantanubhadoria/dist-zilla-pluginbundle-shantanu/is
    sues>. You will be notified automatically of any progress on your issue.

  Source Code
    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    <https://github.com/shantanubhadoria/dist-zilla-pluginbundle-shantanu>

      git clone git://github.com/shantanubhadoria/dist-zilla-pluginbundle-shantanu.git

AUTHOR
    Shantanu Bhadoria <shantanu@cpan.org>

CONTRIBUTORS
    *   Shantanu <shantanu@cpan.org>

    *   Shantanu Bhadoria <shantanu.bhadoria@gmail.com>

    *   Shantanu Bhadoria <shantanu@shantanu-M14xR2.(none)>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2013 by Shantanu Bhadoria.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



CHANGES
Changelog for Dist-Zilla-PluginBundle-SHANTANU

0.14 2013-09-12T06:15:13
 - Renamed CHANGES to Changes [Shantanu]

0.13 2013-09-12T06:07:57
 - Moved to Changelog From git [Shantanu]
 - Bump Changes [Shantanu]

0.12 2013-09-11T03:21:17
 - fixed Deps [Shantanu]
 - Bump Changes [Shantanu Bhadoria]

0.11 2013-07-11T03:37:54
 - v0.11
   
	   - 2013-07-11 11:37:13 Asia/Singapore [Shantanu Bhadoria]
 - Added Perl Version, few fixes to ignore META.yml for git [Shantanu
   Bhadoria]
 - Bump Changes [Shantanu Bhadoria]

0.10 2013-07-11T01:10:14
 - v0.10
   
	   - 2013-07-11 09:03:49 Asia/Singapore [Shantanu Bhadoria]
 - Added Meta YAML generator to help older parsers and KWalitee [Shantanu
   Bhadoria]
 - Bump Changes [Shantanu Bhadoria]

0.09 2013-07-09T13:54:24
 - v0.09
   
	   - 2013-07-09 21:53:25 Asia/Singapore
   
     - Modifications for exclude_filename and exclude_match flags [Shantanu
   Bhadoria]
 - Modification for exclude_filename bug [Shantanu Bhadoria]
 - Bump Changes [Shantanu Bhadoria]

0.08 2013-07-09T10:13:16
 - v0.08
   
	   - 2013-07-09 18:12:37 Asia/Singapore [Shantanu Bhadoria]

0.07 2013-07-09T10:00:24
 - Swapped Dist::Zilla::Plugin::MakeMaker with
   Dist::Zilla::Plugin::MakeMaker::Awesome [Shantanu Bhadoria]
 - Bump Changes [Shantanu Bhadoria]

0.04 2013-03-05T12:37:47
 - v0.04
   
	   - 2013-03-05 20:36:44 Asia/Singapore [Shantanu Bhadoria]
 - Fixed .gitignore to ignore PERLTIDY log [Shantanu Bhadoria]
 - Fixed a spelling mistake [Shantanu Bhadoria]
 - Bump Changes [Shantanu Bhadoria]

0.03 2013-03-05T12:09:35
 - v0.03
   
	   - 2013-03-05 20:08:40 Asia/Singapore [Shantanu Bhadoria]
 - Bump Changes [Shantanu Bhadoria]

0.02 2013-03-05T12:08:16
 - v0.02
   
	   - 2013-03-05 20:07:37 Asia/Singapore [Shantanu Bhadoria]
 - Fixed VERSION Comment [Shantanu Bhadoria]

0.01 2013-03-05T10:55:51
 - Fixed VERSION Comment [Shantanu Bhadoria]
 - Added Changelogfromgit plugin [Shantanu Bhadoria]
 - fixed dist.ini [Shantanu Bhadoria]
 - Modified gitignore to allow META.yml files added README Pod for githubi
   and CHangelog file [Shantanu Bhadoria]
 - Adding Git ignore file [Shantanu Bhadoria]
 - release-0.01 [Shantanu Bhadoria]
 - Initial commit [Shantanu Bhadoria]



lib/Pod/Weaver/PluginBundle/SHANTANU.pm
use strict;
use warnings;

package Pod::Weaver::PluginBundle::SHANTANU;

our $VERSION = '0.19'; # VERSION

use Pod::Weaver 3.101638;
use Pod::Weaver::Config::Assembler;

# Dependencies
use Pod::Weaver::Plugin::WikiDoc ();
use Pod::Elemental::Transformer::List 0.101620 ();
use Pod::Weaver::Section::Support 1.001        ();
use Pod::Weaver::Section::Contributors 0.001   ();

sub _exp { Pod::Weaver::Config::Assembler->expand_package( $_[0] ) }

my $repo_intro = <<'END';
This is open source software.  The code repository is available for
public review and contribution under the terms of the license.
END

my $bugtracker_content = <<'END';
Please report any bugs or feature requests through github at 
{WEB}.
You will be notified automatically of any progress on your issue.
END


sub mvp_bundle_config {
    my @plugins;
    push @plugins, (
        [ '@SHANTANU/WikiDoc',  _exp('-WikiDoc'),  {} ],
        [ '@SHANTANU/CorePrep', _exp('@CorePrep'), {} ],
        [ '@SHANTANU/Name',     _exp('Name'),      {} ],
        [ '@SHANTANU/Version',  _exp('Version'),   {} ],

        [ '@SHANTANU/Prelude', _exp('Region'), { region_name => 'prelude' } ],
        [ '@SHANTANU/Synopsis', _exp('Generic'), { header => 'SYNOPSIS' } ],
        [
            '@SHANTANU/Description', _exp('Generic'),
            { header => 'DESCRIPTION' }
        ],
        [ '@SHANTANU/Overview', _exp('Generic'), { header => 'OVERVIEW' } ],

        [ '@SHANTANU/Stability', _exp('Generic'), { header => 'STABILITY' } ],
    );
    for my $plugin (
        [ 'Usage',        _exp('Collect'), { command => 'usage' } ],
        [ 'Functions',    _exp('Collect'), { command => 'func' } ],
        [ 'Attributes',   _exp('Collect'), { command => 'attr' } ],
        [ 'Constructors', _exp('Collect'), { command => 'construct' } ],
        [ 'Methods',      _exp('Collect'), { command => 'method' } ],
      )
    {
        $plugin->[2]{header} = uc $plugin->[0];
        push @plugins, $plugin;
    }

    push @plugins,
      (
        [ '@SHANTANU/Leftovers', _exp('Leftovers'), {} ],
        [ '@SHANTANU/postlude', _exp('Region'), { region_name => 'postlude' } ],
        [
            '@SHANTANU/Support',
            _exp('Support'),
            {
                perldoc            => 0,
                websites           => 'none',
                bugs               => 'metadata',
                bugs_content       => $bugtracker_content,
                repository_link    => 'both',
                repository_content => $repo_intro
            }
        ],
        [ '@SHANTANU/Authors',      _exp('Authors'),      {} ],
        [ '@SHANTANU/Contributors', _exp('Contributors'), {} ],
        [ '@SHANTANU/Legal',        _exp('Legal'),        {} ],
        [ '@SHANTANU/List', _exp('-Transformer'), { 'transformer' => 'List' } ],
      );

    return @plugins;
}

__END__

=pod

=head1 NAME

Pod::Weaver::PluginBundle::SHANTANU

=head1 VERSION

version 0.19

=head1 METHODS

=head2 mvp_bundle_config

=head1 AUTHOR

Shantanu Bhadoria <shantanu@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Shantanu Bhadoria.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL
# This Makefile.PL for  was generated by Dist::Zilla::Plugin::MakeMaker::Awesome 0.15.
# Don't edit it but the dist.ini used to construct it.

use strict;
use warnings;

use 5.010;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist Zilla Plugin Bundle the way I like to use it",
  "AUTHOR" => "Shantanu Bhadoria <shantanu\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-SHANTANU",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::PluginBundle::SHANTANU",
  "PREREQ_PM" => {
    "Dist::Zilla" => "4.3",
    "Dist::Zilla::Plugin::AutoMetaResources" => 0,
    "Dist::Zilla::Plugin::ChangelogFromGit::CPAN::Changes" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckMetaResources" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::ContributorsFromGit" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::InsertCopyright" => 0,
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker::Awesome" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaNoIndex" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
    "Dist::Zilla::Plugin::MetaTests" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => 0,
    "Dist::Zilla::Plugin::PerlTidy" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
    "Dist::Zilla::Plugin::TaskWeaver" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Kwalitee::Extra" => 0,
    "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
    "Dist::Zilla::Plugin::Test::Version" => 0,
    "Dist::Zilla::PluginBundle::Git" => "2.009",
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Moose" => "0.99",
    "Moose::Autobox" => 0,
    "Pod::Elemental::Transformer::List" => "0.101620",
    "Pod::Weaver" => "3.101638",
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::Section::Contributors" => "0.001",
    "Pod::Weaver::Section::Support" => "1.001",
    "Test::Portability::Files" => "0.06",
    "autodie" => "2.00",
    "namespace::autoclean" => "0.09",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Find" => 0,
    "File::Spec::Functions" => 0,
    "File::Temp" => 0,
    "List::Util" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.19",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



